typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -335,6 +335,34 @@ export class ContractVerifier {
     );
   }
 
+  async getVerifiedContractSourceCode(
+    chain: ChainName,
+    address: Address,
+    verificationLogger: Logger,
+  ): Promise<Partial<ContractVerificationInput>> {
+    verificationLogger.trace(
+      `Fetching contract ABI for ${chain} address ${address}`,
+    );
+    const sourceCodeResults = (
+      await this.submitForm(
+        chain,
+        ExplorerApiActions.GETSOURCECODE,
+        verificationLogger,
+        { address },
+      )
+    )[0]; // This specific query only returns 1 result
+
+    return {
+      address,
+      name: sourceCodeResults.ContractName,
+      expectedimplementation:
+        sourceCodeResults.ImplementationAddress ||
+        sourceCodeResults.Implementation,
+      isProxy:
+        sourceCodeResults.IsProxy === 'true' || sourceCodeResults.Proxy === '1',
+    };
```

Comment:
what happens if the contract is not verified? does this throw an error?

Reply:
yes it will throw an error based on `this.submitForm()`. For example, if there's an api key missing.

Reply:
wdyt of catching the error and returning that the contract is not verified (depending on the error maybe), like I suggested above?

Reply:
yup, i modified this function to do what you suggested

Reply:
it is now called `getContractVerificationStatus()`, which can throw for some arbitrary reason (e.g. missing API key), return isVerification. I let the caller handle the thrown error, in which case, we just return an empty object with an error.